hibernate-core/src/main/java/org/hibernate/jpa/event/internal/CallbackBuilderLegacyImpl.java

```diff
@@ -239,9 +237,22 @@ public void release() {
 
 	@SuppressWarnings({"unchecked", "WeakerAccess"})
 	public Callback[] resolveEmbeddableCallbacks(Class entityClass, Property embeddableProperty, CallbackType callbackType, ReflectionManager reflectionManager) {
-
-		final String embeddableClassName = embeddableProperty.getType().getReturnedClass().getName();
-		final XClass embeddableXClass = reflectionManager.classForName( embeddableClassName );
+		XClass embeddableXClass = null;
+		//First lookup for Collection of embeddable and get the embeddable class
+		XClass entityXClass = reflectionManager.classForName(entityClass.getName());
+		for (XProperty prop : entityXClass.getDeclaredProperties(AccessType.FIELD.getType())) {
+			if (prop.getName().equals(embeddableProperty.getName())) {
+				if(prop.isCollection()){
+					embeddableXClass = prop.getElementClass();
+					break;
+				}
+			}
+		}
+		//else get class from a single embeddable
+		if(embeddableXClass == null){
+			final String embeddableClassName = embeddableProperty.getType().getReturnedClass().getName();
+			embeddableXClass = reflectionManager.classForName( embeddableClassName );
+		}
```

Comment:
again, we need more aggressive delimiter usage. Please refer to existing code for the pattern.

Reply:
````
if(embeddableXClass == null){
````

is fugly in an readability way.  Please use spaces appropriately.  

````
if ( embeddableXClass == null ) {
````

would be more consistent with the rest of the codebase, although 

````
if (embeddableXClass == null) {
````

is fine as well.

Is this what you meant by "delimiter usage" @NathanQingyangXu ?

Reply:
Yeah. Such format issue often troubles new contributor (including me). Is it an idea to add the link of https://github.com/hibernate/hibernate-ide-codestyles into our wiki article (i.e. https://github.com/hibernate/hibernate-orm/wiki/Contributing-Code)?

Reply:
After testing, I found the style file cannot be imported into the latest Intellij IDEA (2019.3), :(